File: app/dtos/notification.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict

from app.models.notifications import NotificationType


class NotificationResponse(BaseModel):
    id: int
    notification_type: NotificationType
    target_type: str
    target_id: int
    message: str
    is_read: bool
    created_at: datetime
    actor_nickname: str | None = None

    model_config = ConfigDict(from_attributes=True)

File: web/src/types/notification.ts
export type NotificationType = "COMMENT" | "REPLY" | "LIKE" | "CHALLENGE_KICK" | "CHALLENGE_INVITE" | "CHALLENGE_DELETE" | "RISK_CHANGE";

export interface Notification {
  id: number;
  notification_type: NotificationType;
  target_type: "POST" | "COMMENT" | "VERIFICATION" | "CHALLENGE" | "DISEASE_RISK";
  target_id: number;
  message: string;
  is_read: boolean;
  created_at: string;
  actor_nickname: string | null;
}
